vue-app/src/lambda/recipient.js

```diff
@@ -0,0 +1,68 @@
+import { GoogleSpreadsheet } from 'google-spreadsheet'
+
+/**
+ * Creates an object row from a RecipientApplicationData. A row is a key-value
+ * object where the keys are composed by the concatenation of the nested keys.
+ *
+ * From:
+ * const recipient = {
+ *    project: {
+ *      name: "test",
+ *      tagline: "string",
+ *      description: "string",
+ *      category: "string",
+ *      problemSpace: "string",
+ *    },
+ *    ...
+ * };
+ *
+ * To:
+ * const row = { projectname: "test", projecttagline: "string", ... }
+ */
+function recipientToRow(recipient) {
+  return Object.keys(recipient).reduce((result, header) => {
+    if (typeof recipient[header] !== 'object') {
+      return { ...result, [header]: recipient[header] }
+    }
+
+    const childObj = Object.keys(recipient[header]).reduce(
+      (result, subheader) => {
+        const key = header + subheader
+        return { ...result, [key]: recipient[header][subheader] }
+      },
+      {}
+    )
+
+    return { ...result, ...childObj }
+  }, {})
+}
+
+exports.handler = async function (event) {
+  if (!event.body) {
+    return {
+      statusCode: 400, // Bad request
+    }
+  }
+
+  try {
+    const recipient = JSON.parse(event.body)
+
+    const creds = JSON.parse(process.env.VUE_APP_GOOGLE_APPLICATION_CREDENTIALS)
+    const doc = new GoogleSpreadsheet(process.env.VUE_APP_GOOGLE_SPREADSHEET_ID)
```

Comment:
If I am correct, since we're calling this from a lambda function, and not the UI, I believe we can change these variables to just `GOOGLE_APPLICATION_CREDENTIALS` and `GOOGLE_SPREADSHEET_ID` and it will prevent them from being exposed to the vue-app front end.

We should update the `vue-app/.env.example` with these variables too

Reply:
Great catch! haha yes, it was throwing away the whole purpose of this lambda function (hiding the credentials). Thanks